### PR TITLE
[SMP] Add an experiment with higher DogStatsd string interner size

### PR DESCRIPTION
**Pull request by blt**
### What does this PR do?

This commit introduces an environment variable that raises the DogStatsd string interner size from its default of 4096 to 20480. This experiment is a variant of the other no distribution, no multivalue experiments where we observe that stringInterner is a source of CPU and allocation consumption.

REF SMPTNG-12
REF #19179

### CODE REVIEW

File: test/regression/cases/uds_dogstatsd_to_api_nodist_nomulti_200MiB_interner_20480/experiment.yaml
Comment (GeorgeHahn) [inline test/regression/cases/uds_dogstatsd_to_api_nodist_nomulti_200MiB_interner_20480/experiment.yaml:1]: Could you add a comment to this file to note that it is a short-lived experiment to investigate the string interner behavior?

**Review by GeorgeHahn — CHANGES_REQUESTED** [review]

File: test/regression/cases/uds_dogstatsd_to_api_nodist_nomulti_200MiB_interner_20480/experiment.yaml
Comment (blt) [reply test/regression/cases/uds_dogstatsd_to_api_nodist_nomulti_200MiB_interner_20480/experiment.yaml:1]: Oh yeah, sure thing. Solid idea.

**Comment by blt** [comment]
Interesting results. Compare the [profiles](https://app.datadoghq.com/profiling/comparison?query=account%3Asingle-machine-performance%20runtime%3Ago%20client_team%3Aagent%20experiment%3Auds_dogstatsd_to_api_nodist_nomulti_200mib_interner_20480&compare_end_A=1696042416025&compare_end_B=1696044054384&compare_query_A=account%3Asingle-machine-performance%20runtime%3Ago%20client_team%3Aagent%20experiment%3Auds_dogstatsd_to_api_nodist_nomulti_200mib&compare_start_A=1696022987474&compare_start_B=1696022241502&compareValuesMode=absolute&my_code=disabled&profile_type=alloc-samples&viz=flame_graph&start=1696001290638&end=1696087690638&paused=false) and it appears that a higher capacity `stringInterner`

* experiment total allocations drop from 135M to 71M
* live heap drops from 741MiB to 623MiB
* mutex time drops from 10s to 4.3s
* throughput [improves](https://app.datadoghq.com/dashboard/t3q-y5d-i7u/smp-regression-detector-capture-data?refresh_mode=paused&tpl_var_experiment%5B0%5D=uds_dogstatsd_to_api_nodist_nomulti_200mib_interner_20480&tpl_var_experiment%5B1%5D=uds_dogstatsd_to_api_nodist_nomulti_200mib&tpl_var_run-id%5B0%5D=d120b0dc-071f-4510-80f4-da6cac5272ce&from_ts=1696031225413&to_ts=1696031776702&live=false) by 5MiB/s
* CPU utilization drops by 50% 

This strongly suggests that we should pursue:

* a PR that raises the default size
* re-introduction of https://github.com/DataDog/datadog-agent/pull/19179
* potentially a better eviction strategy

Note that we will need to merge this experiment in before making any of the above modifications; we want to be sure the result is an improvement on the low-end of throughput. As such, come Monday I'll introduce a PR that bumps the default. If that's acceptable across the board that will make this PR obsolete and I may close it without merging.

**Comment by GeorgeHahn** [comment]
We should be careful that we're not overfitting agent to this experiment. I find it curious that `(*stringInterner).LoadOrStore` doesn't show up at all in the 20480 experiment. It's possible that we're interning all of the strings in the experiment such that the string cache is never flushed.

Could you re-run this while capturing target metrics for the interner? Those are here: https://github.com/DataDog/datadog-agent/blob/main/comp/dogstatsd/server/intern.go#L18-L29 (Note the histogram isn't supported by lading.)

**Pull request closed without merging**